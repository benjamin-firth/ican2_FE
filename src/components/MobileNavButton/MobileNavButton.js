import React from 'react';
import { Link } from "react-router-dom";
import { useDispatch } from 'react-redux';
import { checkSelected } from '../../utils/methods';
import './MobileNavButton.scss';
import { logoutCurrentUser } from '../../actions';
import PropTypes from 'prop-types';

const MobileNavButton = ({ nav, name }) => {
  const dispatch = useDispatch();

  const logOut = () => {
    if (name === 'log out') {
      dispatch(logoutCurrentUser());
    }
  }

  return (
    <Link to={nav}>
      <button value={name} id='logoutButton' className='mobile-nav-button' onClick={() => logOut()}>
        {name}
      </button>
    </Link>
  );
};

MobileNavButton.propTypes = {
  nav: PropTypes.string,
  name: PropTypes.string
}

export default MobileNavButton;
