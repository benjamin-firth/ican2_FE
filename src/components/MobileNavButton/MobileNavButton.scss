@import '../../variables.scss';

.mobile-nav-button {
  width: 40vw;
  padding: 3vw;
  border: none;
  background-color: white;
  color: $secondarycolor;
  font-size: 5.5vw;
  font-family: $secondaryfont;
  margin: 1vw;
  &:hover {
    background-color: $primarycolor;
    color: white;
    color: $secondarycolor;
  }
}
