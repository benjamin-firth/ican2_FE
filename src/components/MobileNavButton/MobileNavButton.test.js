import React from 'react';
import { shallow } from 'enzyme';
import MobileNavButton from './MobileNavButton';

jest.mock("react-redux", () => ({
  useDispatch: () => jest.fn()
}));

describe('MobileNavButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <MobileNavButton nav='/' name='home' />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire logout', () => {
    wrapper.find('#logoutButton').simulate('click');

    expect(wrapper.find('#logoutButton').prop('value')).toEqual('home');
  });

});
