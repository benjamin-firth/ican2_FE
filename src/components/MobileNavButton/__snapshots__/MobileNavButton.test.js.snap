// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavButton should match the snapshot 1`] = `
<Link
  to="/"
>
  <button
    className="selected nav-button"
    id="logoutButton"
    onClick={[Function]}
    value="home"
  >
    home
  </button>
</Link>
`;
